spark/src/main/java/com/netflix/iceberg/spark/source/Writer.java

```diff
@@ -96,7 +87,8 @@
 
   @Override
   public DataWriterFactory<InternalRow> createWriterFactory() {
-    return new WriterFactory(table.spec(), format, dataLocation(), table.properties(), fileIo);
+    return new WriterFactory(
+        table.spec(), format, table.locationProvider(), table.properties(), fileIo);
```

Comment:
Why not just pass `table` and let `WriterFactory` extract all the required fields from it?

Reply:
Serialization?

Reply:
Yeah, Serialization. WriterFactory has to be Serializable and Table isn't Serializable. We could pass Table to its constructor, assuming that the implementation knows not to keep a reference to it, but I think it is cleaner and more maintainable to not pass non-serializable objects in.

Reply:
Ah I misread sorry, thought this was at the top level of `Writer` - this is correct then.